src/org/apache/commons/httpclient/HttpMethodBase.java

```diff
@@ -1291,17 +1298,17 @@ public boolean validate() {
      * @return cookie spec
      */
     private CookieSpec getCookieSpec(final HttpState state) {
-    	if (this.cookiespec == null) {
-    		int i = state.getCookiePolicy();
-    		if (i == -1) {
-        		this.cookiespec = CookiePolicy.getCookieSpec(this.params.getCookiePolicy());
-    		} else {
-        		this.cookiespec = CookiePolicy.getSpecByPolicy(i);
-    		}
-    		this.cookiespec.setValidDateFormats(
-            		(Collection)this.params.getParameter(HttpMethodParams.DATE_PATTERNS));
-    	}
-    	return this.cookiespec;
+        if (this.cookiespec == null) {
```

Comment:
This changes could be left out (just to keep the changes to the forked class to a minimum, just essential changes).